team-fingerprints-web/src/components/Modals/ModalConfirm/__tests__/ModalConfirm.spec.tsx

```diff
@@ -0,0 +1,52 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { createRef } from "react";
+import { act } from "react-dom/test-utils";
+import ModalConfirm, { IModalConfirmMethods } from "../ModalConfirm";
+
+describe("ModalConfirm", () => {
+  const onConfirmMock = jest.fn();
+
+  it("does trigger onConfirm function and hide the modal when click on confirm button", async () => {
+    const ref = createRef<IModalConfirmMethods>();
+
+    render(
+      <ModalConfirm
+        onConfirm={onConfirmMock}
+        modalMessage="Modal message"
+        ref={ref}
+      />
+    );
+
+    await act(() => ref.current?.show());
+
+    const confirmButton = screen.getByRole("button", { name: "Confirm" });
+
+    await userEvent.click(confirmButton);
+
+    expect(screen.queryByRole("div", { hidden: true })).not.toBeInTheDocument();
+    expect(onConfirmMock).toHaveBeenCalled();
+  });
+
+  it("does hide the modal on cancel click", async () => {
+    const ref = createRef<IModalConfirmMethods>();
+
+    render(
+      <ModalConfirm
+        onConfirm={onConfirmMock}
+        modalMessage="Modal message"
+        ref={ref}
+      />
+    );

```

Comment:
I would suggest taking the `ref` with `onConfirmMock` and `render` method outside to the top of your file because it's all unnecessarily duplicated

So on top of your tests you would have:
```
const onConfirmMock = jest.fn();
const ref = createRef<IModalConfirmMethods>();

const renderComponent = () => {
  render(
    <ModalConfirm
      onConfirm={onConfirmMock}
      modalMessage="Modal message"
      ref={ref}
    />
  );
};
```

And then inside your test:

```suggestion
    renderComponent();
```

Reply:
Right, done